### PR TITLE
[Enhance][Theme] Color @dooboo/theme  colors enhancement

### PR DESCRIPTION
**Pull request by s-ong-c**
## Description

Provide optimized color code for Dooboo UI, like font, background, border
I think it can be used in various ways.

Naming Description: Numbers indicate color intensity.

## Checklist

Before you create this PR confirms that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide](https://github.com/dooboolab/dooboo-ui/blob/master/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] Run `yarn test` or `yarn test -u` if you need to update snapshot.
- [x] Run `yarn lint`
- [x] I am willing to follow-up on review comments in a timely manner.

### CODE REVIEW

File: packages/theme/theme.ts [hunk 1]
```diff
@@ -44,6 +114,37 @@ export const dark = {
...
+  grayBorder0: colors.gray1,
+  grayBorder1: colors.gray2,
+  grayBorder2: colors.gray4,
+  grayBorder3: colors.gray5,
+  disableColor: colors.white2,
+  warnBackground: colors.red0,
+  warnColor: colors.red4,
+  tealBackGround: colors.green0,
+  tealColor: colors.green1,
+  labelInput: {
```
Comment (devethan) [inline packages/theme/theme.ts]: Why does only this part have a different structure?
I think `labelInput.border` can be changed to `border` or `whiteBorder`
`labelInput.label` can be `whiteText`, background can be `darkBackground`

If you expect it'll make an error, you can add a comment about this with remain this code.

Comment (s-ong-c) [reply packages/theme/theme.ts]: I think labelInput is a well-formed structure, so I specified the default lableInput.

But looking again, I don't need to do this only for labelInput. Thanks for the comments

File: packages/theme/theme.ts [hunk 2]
```diff
@@ -44,6 +110,33 @@ export const dark = {
...
+  grayText: colors.white1,
+  line: colors.white1,
+  darkLine: colors.gray5,
+  thickLine: colors.primary5,
+  mediumLine: colors.primary8,
+  border: '#EDEDED',
+  grayBorder0: colors.gray1,
+  grayBorder1: colors.white0,
+  grayBorder2: colors.gray4,
+  grayBorder3: colors.gray5,
```
Comment (hyochan) [inline packages/theme/theme.ts]: Would there be more explicit naming for this? It's hard to tell how to use `0` or `2` at first glance.

File: packages/theme/index.tsx [hunk 3]
```diff
@@ -8,11 +11,13 @@ import {
   light as lightTheme,
 } from './theme';
 import React, { useState } from 'react';
+
 import createCtx from './createCtx';
 
 interface Context {
   themeType: ThemeType;
   theme: DefaultTheme;
+  dooboo: DoobooTheme;
```
Comment (hyochan) [inline packages/theme/index.tsx]: The explicit naming for this would be `doobooTheme` instead of `dooboo`.

File: packages/theme/theme.ts (hunk 2 above)
Comment (s-ong-c) [reply packages/theme/theme.ts]: @hyochan  The intention is that grayBorder 0,1,2 numbers express the degree of darkness.

Comment (devethan) [reply packages/theme/theme.ts]: Screen Shot 2020-07-21 at 11 36 16 AM

[ref](https://yeun.github.io/open-color/)

Comment (hyochan) [reply packages/theme/theme.ts]: How about something like borderLight0, 1, 2...

Comment (s-ong-c) [reply packages/theme/theme.ts]: That looks better
I will try it

**Review by hyochan — APPROVED** [review]

**Pull request merged**